src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions/src/RazorExtensions.cs

```diff
@@ -37,7 +37,9 @@ public static void Register(RazorProjectEngineBuilder builder)
             builder.Features.Add(new PagesPropertyInjectionPass());
             builder.Features.Add(new ViewComponentTagHelperPass());
             builder.Features.Add(new RazorPageDocumentClassifierPass());
-            builder.Features.Add(new MvcViewDocumentClassifierPass());
+
+            builder.Features.Add(new ConsolidatedMvcViewDocumentClassifierPass(builder.Configuration.LanguageVersion));
```

Comment:
Could we accept an option here? `bool generateConsolidatedViews` or something like that?